libc/newhdrgen/class_implementation/classes/function.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+#
+# ====-- Function class for libc function headers -------------*- python -*--==#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ==-------------------------------------------------------------------------==#
+
+
+class Function:
+    def __init__(
+        self, standards, return_type, name, arguments, guard=None, attributes=[]
+    ):
+        self.standard = standards
+        self.return_type = return_type
+        self.name = name
+        self.arguments = [arg["type"] for arg in arguments]
```

Comment:
Why do we extract just the type for the argument? Don't we need the identifier? Parameter attributes?

What fields are you expecting an `arg` to have?

Reply:
The way we use arguments in the Yaml file is carrying the parameter type so we can identify it as type in arguments. 
```
- name: call_once
    standard: stdc
    return_type: void
    arguments:
      - type: once_flag *
      - type: __call_once_func_
```

Reply:
We also have this in case we end up having to separate the argument type into prefix, name, suffix. If it is confirmed that we don't need to do so, we can easily remove "type:" so that it is a regular list.

Reply:
If we end up adding more fields to an `argument`, then we might want just to keep a `Function`'s `arguments` as a list of dict's rather than list of strings of the parameter types.  But I'll leave that up to later patches that make use of these objects.